tests/test_tv.py

```diff
@@ -0,0 +1,136 @@
+# coding=utf-8
+"""Tests for medusa.tv.py."""
+
+from medusa.common import ARCHIVED, DOWNLOADED, Quality, SNATCHED, SNATCHED_PROPER, WANTED
+from medusa.tv import TVEpisode
+import pytest
+
+
+@pytest.mark.parametrize('p', [
+    {  # p0: Downloaded 720p HDTV and found 720p BluRay: yes
+        'ep_status': DOWNLOADED,
+        'cur_quality': Quality.HDTV,
+        'new_quality': Quality.HDBLURAY,
+        'allowed_qualities': [Quality.HDTV, Quality.HDWEBDL, Quality.HDBLURAY],
+        'preferred_qualities': [],
+        'expected': True
```

Comment:
this is 'no' right?

when user sets only allowed_qualities medusa downloads only the first available. it doesn't re-download.